compiler/rustc_mir_transform/src/lib.rs

```diff
@@ -604,6 +602,7 @@ fn run_optimization_passes<'tcx>(tcx: TyCtxt<'tcx>, body: &mut Body<'tcx>) {
             &dead_store_elimination::DeadStoreElimination::Initial,
             &gvn::GVN,
             &simplify::SimplifyLocals::AfterGVN,
+            &match_branches::MatchBranchSimplification,
```

Comment:
Should this run on clone shims too? Or do we want a trait-based solution to detect trivial clone impls?